vowpalwabbit/cb_label_parser.h

```diff
@@ -68,15 +63,25 @@ char* bufcache_label(LabelT& ld, char* c)
     *(LabelElmT*)c = ld.costs[i];
     c += sizeof(LabelElmT);
   }
-  return bufcache_label_additional_fields(ld, c);
+
+  return c;
+}
+
+template <typename LabelT = CB::label>
+void cache_label_additional_fields(LabelT& ld, io_buf& cache)
+{
+  cache.write_value(ld.weight);
 }
 
 template <typename LabelT = CB::label, typename LabelElmT = cb_class>
 void cache_label(LabelT& ld, io_buf& cache)
 {
   char* c;
   cache.buf_write(c, sizeof(size_t) + sizeof(LabelElmT) * ld.costs.size());
```

Comment:
`cache.buf_write(c, sizeof(size_t) + sizeof(LabelElmT) * ld.costs.size());` should probably be done internally to the call of `bufcache_label_elements`, since it is overloaded based on type for continuous right?

Reply:
I believe the point is that it computes the right thing based on that `sizeof()` (which is what the change in the Continuous Action commit did) - since `LabelElmT` is the POD containing just the element data, so it matches the read/write size. The only place where they differ is that the continuous label does not have additional fields (weight).

Reply:
That's not obvious at all but I see what you mean. I don't see any benefit of doing it that way over just delegating responsibility of requesting enough buffer space to the `bufcache_label_elements` call itself.

Reply:
It seems like we should pull `bufcache_label_elements` back into `cache_label`, since it never gets overloaded anyways.

Reply:
I pulled in the element cache/read methods into the main label cache/read methods, and replaced most of the explicit io_buf buffer management with `[read/write]_value`, but there is an odd lack-symmetry around label caching forcing the read to be able to fail transparently when at EOF.